app/controllers/jobs_controller.rb

```diff
@@ -0,0 +1,41 @@
+class JobsController < ApplicationController
+  before_action :load_job
+
+  def show
+    if @job.completed?
+      show_completed_job
+    elsif @job.errored?
+      show_errored_job
+    else
+      show_submitted_job
+    end
+  end
+
+  protected
+
+  def publicly_accessible?
+    true
+  end
+
+  private
+
+  def load_job
+    @job ||= TrackableJob::Job.find(params[:id])
+  end
+
+  def show_completed_job
+    redirect_to @job.redirect_to if @job.redirect_to.present?
```

Comment:
yup, then we will return that to the client, and then the client will download that.